### PR TITLE
Integrate WC_Order_Query

### LINKED ISSUE #14795: Integrate WC_Order_Query
Use `WC_Order_Query` in `wc_get_orders`, `WC_Order_Data_Store_CPT::get_orders`, and anywhere else it makes sense.

See #14733

### PR DESCRIPTION
**Pull request by claudiulodro**
Closes #14795.

I've deprecated `WC_Order_Data_Store_CPT::get_orders` on the off chance that someone is using it for something already.

This PR also includes a tweak to `parse_date_for_wp_query` that works around some WordPress limitations.

### CODE REVIEW

File: includes/data-stores/class-wc-order-data-store-cpt.php [hunk 1]
```diff
@@ -684,11 +624,16 @@ protected function get_wp_query_args( $query_vars ) {
...
+			$values = is_array( $query_vars['customer'] ) ? $query_vars['customer'] : array( $query_vars['customer'] );
+			$wp_query_args['meta_query'][] = $this->get_orders_generate_customer_meta_query( $values );
+		}
+
 		if ( ! isset( $query_vars['paginate'] ) || ! $query_vars['paginate'] ) {
 			$wp_query_args['no_found_rows'] = true;
 		}
 
-		return apply_filters( 'woocommerce_get_order_wp_query_args', $wp_query_args, $query_vars );
+		return apply_filters( 'woocommerce_order_data_store_cpt_get_orders_query', $wp_query_args, $query_vars, $this );
```
Comment (claudiulodro) [inline includes/data-stores/class-wc-order-data-store-cpt.php]: This is the same filter name that `::get_orders` used.

File: includes/abstracts/abstract-wc-object-query.php [hunk 2]
```diff
@@ -71,8 +71,8 @@ protected function get_default_query_vars() {
 			'parent_exclude' => '',
 			'exclude'        => '',
 
-			'limit'          => -1,
-			'page'           => '',
+			'limit'          => get_option( 'posts_per_page' ),
```
Comment (mikejolley) [inline includes/abstracts/abstract-wc-object-query.php]: Just curious why this changed from unlimited to posts per page?

File: includes/data-stores/class-wc-order-data-store-cpt.php [hunk 3]
```diff
@@ -684,11 +624,16 @@ protected function get_wp_query_args( $query_vars ) {
 			}
 		}
 
+		if ( isset( $query_vars['customer'] ) && '' !== $query_vars['customer'] ) {
```
Comment (mikejolley) [inline includes/data-stores/class-wc-order-data-store-cpt.php]: Wouldn't `! empty` would here?

File: includes/wc-order-functions.php [hunk 4]
```diff
@@ -13,65 +13,26 @@
...
- *      orderby string Order by date, title, id, modified, rand etc
- *      order string ASC or DESC
- *      return string Type of data to return. Allowed values:
- *          ids array of order ids
- *          objects array of order objects (default)
- *      paginate bool If true, the return value will be an array with values:
- *          'orders'        => array of data (return value above),
- *          'total'         => total number of orders matching the query
- *          'max_num_pages' => max number of pages found
+ * Args and usage: @todo link to wiki page.
```
Comment (mikejolley) [inline includes/wc-order-functions.php]: ;)

File: includes/wc-order-functions.php [hunk 5]
```diff
@@ -82,7 +43,29 @@ function wc_get_orders( $args ) {
...
-	return WC_Data_Store::load( 'order' )->get_orders( $args );
+	// Map legacy date args to modern date args.
+	$date_before = false;
+	$date_after  = false;
+
+	if ( ! empty( $args['date_before'] ) ) {
+		$datetime    = wc_string_to_datetime( $args['date_before'] );
+		$date_before = strpos( $args['date_before'], ':' ) ? $datetime->getOffsetTimestamp() : $datetime->date( 'Y-m-d' );
+	}
+	if ( ! empty ( $args['date_after'] ) ) {
```
Comment (mikejolley) [inline includes/wc-order-functions.php]: There is a space before bracket

File: includes/abstracts/abstract-wc-object-query.php (hunk 2 above)
Comment (mikejolley) [inline includes/abstracts/abstract-wc-object-query.php]: Ah to match the old defaults in wc_get_orders 👍

**Review by mikejolley — COMMENTED** [review]
If this passes unit tests, all good :) Few notes and minor things.

File: includes/data-stores/class-wc-order-data-store-cpt.php (hunk 3 above)
Comment (claudiulodro) [inline includes/data-stores/class-wc-order-data-store-cpt.php]: @mikejolley Apparently not. :) [Looks like 0 is a valid param](https://github.com/woocommerce/woocommerce/blob/master/tests/unit-tests/order/functions.php#L166). I've switched it to check for empty string and empty array, though.

**Pull request merged**